File: TiaCodegen/Commands/Distributor.cs
using TiaCodegen.Interfaces;

namespace TiaCodegen.Commands
{
    public class Distributor : BaseOperationOrSignal
    {
        public Distributor(params IOperationOrSignal[] operationOrSignals)
            : base(operationOrSignals)
        { }

        public override int CreateContactAndFillCardinality(IOperationOrSignal parent)
        {
            int c = 0;
            this.OperationId = parent.OperationId;
            foreach (var op in Children)
            {
                c += op.CreateContactAndFillCardinality(this);
            }

            this.Cardinality = c;
            return this.Cardinality;
        }
    }
}

File: TiaCodegen-ts/src/Commands/Distributor.ts
import { IOperationOrSignal } from '../Interfaces/IOperationOrSignal.js';
import { BaseOperationOrSignal } from './BaseOperationOrSignal.js';

export class Distributor extends BaseOperationOrSignal {
    constructor(...operationOrSignals: IOperationOrSignal[]) {
        super(...operationOrSignals);
    }

    createContactAndFillCardinality(parent: IOperationOrSignal): number {
        let c = 0;
        this.operationId = parent.operationId;
        for (const op of this.children) {
            c += op.createContactAndFillCardinality(this);
        }
        this.cardinality = c;
        return this.cardinality;
    }
}
